sacc/tracers.py

```diff
@@ -198,13 +214,302 @@ def from_tables(cls, table_list):
+            tables.append(table)
+        return tables
+
+    @classmethod
+    def from_tables(cls, table_list):
+        tracers = {}
+
+        # Collect beam and bandpass tables describing the same tracer
+        tr_tables = {}
+        for table in table_list:
+            # Read name and table type
+            name = table.meta['SACCNAME']
+            quantity = table.meta['SACCQTTY']
+            tabtyp = table.meta['EXTNAME'].split(':')[-1]
+            if tabtyp not in ['beam']:
+                raise KeyError("Unknown table type " + table.meta['EXTNAME'])
+
+            # If not present yet, create new tracer entry
+            if name not in tr_tables:
+                tr_tables[name] = {}
+            # Add table
+            tr_tables[name][tabtyp] = table
+
+        # Now loop through different tracers and build them from their tables
+        for n, dt in tr_tables.items():
+            quantity = []
+            metadata = {}
+            map_unit = 'none'
+            ell = []
+            beam_ell = []
+            beam_extra = {}
+            spin = 0
+
+            if 'beam' in dt:
+                table = dt['beam']
+                name = table.meta['SACCNAME']
+                quantity = table.meta['SACCQTTY']
+                ell = table['ell']
+                beam_ell = table['beam_ell']
+                for col in table.columns.values():
+                    if col.name not in ['ell', 'beam_ell']:
+                        beam_extra[col.name] = col.data
+                map_unit = table.meta['MAP_UNIT']
+                spin = table.meta['SPIN']
+                for key, value in table.meta.items():
+                    if key.startswith("META_"):
+                        metadata[key[5:]] = value
+
+            tracers[name] = cls(name, quantity, spin, ell, beam_ell,
+                                beam_extra=beam_extra,
+                                map_unit=map_unit, metadata=metadata)
+        return tracers
+
+
+class NuMapTracer(BaseTracer, tracer_type='NuMap'):
+    """
+    A Tracer type for a sky map at a given frequency.
+
+    Takes at least four arguments, defining the bandpass and beam.
+
+    Attributes
+    ----------
+
+    name: str
+        The name for this specific tracer, e.g. a frequency band
+        identifier.
+    quantity: str
+        String describing the physical quantity described by this
+        tracer (e.g. intensity, antenna_temperature, cmb_temperature,
+        HI_temperature).
+    spin: int
+        Spin for this observable. Either 0 (e.g. intensity)
+        or 2 (e.g. polarization).
+    nu: array
+         Array of frequencies.
+    bpss_nu: array
+         Bandpass transmission.
+    bpss_extra: array
+         Other bandpass-related arrays
+         (e.g. uncertainties, principal components,
+         alternative measurements, whatever).
+    ell: array
+         Array of multipole values at which the beam is defined.
+    beam_ell: array
+         Beam.
```

Comment:
I guess you assume the beam is separable in ell and nu? This is likely not the case in detail IIUIC. It may be an appropriate approximation.

Reply:
This is what is assumed in all existing experiments that I know of (Planck, ACT, BICEP). I agree that it's not true in detail, and it may be that this needs to be generalized in the future, but I'd rather address that later.